servicetalk-loadbalancer-experimental-provider/src/main/java/io/servicetalk/loadbalancer/experimental/DefaultHttpLoadBalancerProvider.java

```diff
@@ -0,0 +1,119 @@
+        final String serviceName = clientNameFromAddress(address);
+        if (config.enabledForServiceName(serviceName)) {
+            try {
+                HttpLoadBalancerFactory<R> loadBalancerFactory = DefaultHttpLoadBalancerFactory.Builder.<R>from(
+                        defaultLoadBalancer(serviceName)).build();
+                builder = builder.loadBalancerFactory(loadBalancerFactory);
+                return new LoadBalancerIgnoringBuilder(builder, serviceName);
+            } catch (Throwable ex) {
+                LOGGER.warn("Failed to enabled DefaultLoadBalancer for client to address {}.", address, ex);
+            }
+        }
+        return builder;
+    }
+
+    private <R> LoadBalancerFactory<R, FilterableStreamingHttpLoadBalancedConnection> defaultLoadBalancer(
+            String serviceName) {
+        return LoadBalancers.<R, FilterableStreamingHttpLoadBalancedConnection>
+                        builder("experimental-load-balancer")
+                .loadBalancerObserver(new DefaultLoadBalancerObserver(serviceName))
+                // set up the new features.
+                .outlierDetectorConfig(config.outlierDetectorConfig())
+                .loadBalancingPolicy(config.getLoadBalancingPolicy())
+                .build();
+    }
+
+    /**
+     * Extract the service name from the address object.
+     * Note: this is a protected method to allow overriding for custom address types.
+     * @param <U> the unresolved type of the address.
+     * @param address the address from which to extract the service name.
+     * @return the String representation of the provided address.
+     */
+    protected <U> String clientNameFromAddress(U address) {
+        String serviceName;
+        if (address instanceof HostAndPort) {
+            serviceName = ((HostAndPort) address).hostName();
+        } else if (address instanceof String) {
+            serviceName = (String) address;
+        } else {
+            LOGGER.warn("Unknown service address type={} was provided, "
+                    + "default 'toString()' will be used as serviceName", address.getClass());
+            serviceName = address.toString();
+        }
+        return serviceName;
+    }
+
+    private static final class LoadBalancerIgnoringBuilder<U, R>
+            extends DelegatingSingleAddressHttpClientBuilder<U, R> {
+
+        private final String serviceName;
+
+        LoadBalancerIgnoringBuilder(final SingleAddressHttpClientBuilder<U, R> delegate, final String serviceName) {
+            super(delegate);
+            this.serviceName = serviceName;
+        }
+
+        @Override
+        public SingleAddressHttpClientBuilder<U, R> loadBalancerFactory(
+                HttpLoadBalancerFactory<R> loadBalancerFactory) {
+            LOGGER.info("Ignoring load balancer factory for client to {} which has DefaultLoadBalancer enabled.",
```

Comment:
consider also including the loadBalancerFactory class name, probably useful for troubleshooting

Reply:
Will do. I imagine it's nearly always going to be `DefaultHttpLoadBalancerFactory` but you never know.